tests/qos/qos_sai_base.py

```diff
@@ -945,10 +945,22 @@ def dutAsicConfig(self, asic, duthost):
                     if line:
                         m = re.match('THDI_BUFFER_CELL_LIMIT_SP\(0\).*\<LIMIT=(\S+)\>', line)
                         if m:
-                            asicConfig['shared_limit_sp0'] = int(m.group(1), 0)
+                            asicConfig['ingress_shared_limit_sp0'] = int(m.group(1), 0)
                             break
+
+                output = duthost.shell('bcmcmd "g MMU_THDM_DB_POOL_SHARED_LIMIT"', module_ignore_errors=True)
+                logger.info('Read ASIC MMU_THDM_DB_POOL_SHARED_LIMIT register, output {}'.format(output))
+                count = 0
+                for line in output['stdout'].replace('\r', '\n').split('\n'):
+                    if line:
+                        m = re.match('MMU_THDM_DB_POOL_SHARED_LIMIT\(([01])\).*\]\=(\S+)', line)
+                        if m:
+                            asicConfig['egress_shared_limit_sp{}'.format(m.group(1))] = int(m.group(2), 0)
+                            count += 1
+                            if count == 2:
```

Comment:
Why break after count equals 2? Only 2 SPs?

Reply:
yes, one is for lossy, another one is for lossless